### PR TITLE
Kafka hello-world

### PR DESCRIPTION
**Pull request by LucyxxNie**

### CODE REVIEW

File: src/kafka/kafka_hello_world.clj [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+(defn build-consumer
+  [consumer-properties topic]
+  (let [consumer        (KafkaConsumer. consumer-properties)
+        subscribe-topic (.subscribe consumer [topic])]
+    consumer))
+
+(defn new-msg
+  [topic key msg]
+  (ProducerRecord. topic key msg))
```
Comment (lukaszhall) [inline src/kafka/kafka_hello_world.clj]: Same comment as before

File: src/kafka/kafka_hello_world.clj [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+(defn send-msg!
+  [producer msg-value]
+  (let [producer-configuration "example-topic"
+        msg                    (new-msg producer-configuration "msg" msg-value)
+        sending-msg            (.send producer msg)]
```
Comment (lukaszhall) [inline src/kafka/kafka_hello_world.clj]: name?

File: src/kafka/kafka_hello_world.clj [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  [producer-properties]
+  (KafkaProducer. producer-properties))
+
+(defn build-consumer
+  [consumer-properties topic]
+  (let [consumer        (KafkaConsumer. consumer-properties)
+        subscribe-topic (.subscribe consumer [topic])]
+    consumer))
+
+(defn new-msg
```
Comment (lukaszhall) [inline src/kafka/kafka_hello_world.clj]: `new` is an oop term

File: src/kafka/kafka_hello_world.clj [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        sending-msg            (.send producer msg)]
+    (println "Msg sent successfully!")
+    sending-msg))
+
+(defn receive-msg
+  [consumer]
+  (let [records (.poll consumer (Duration/ofMillis 100000))]
+    (->> records
+         (mapv (fn [record]
+                 (str (.value record)))))))
```
Comment (lukaszhall) [inline src/kafka/kafka_hello_world.clj]: Same as before... _**Do not put code outside of the binding blocks**_

File: src/kafka/kafka_hello_world.clj (hunk 1 above)
Comment (LucyxxNie) [reply src/kafka/kafka_hello_world.clj]: I saw you commented on this, but it went away when I refresh the page, can you leave the comment again?

File: src/kafka/kafka_hello_world.clj [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+(defn send-msg!
+  [producer msg-value]
+  (let [producer-configuration "example-topic"
+        msg                    (new-msg producer-configuration "msg" msg-value)
+        sending-msg            (.send producer msg)]
+    (println "Msg sent successfully!")
+    sending-msg))
+
+(defn receive-msg
```
Comment (lukaszhall) [inline src/kafka/kafka_hello_world.clj]: is this appropriately named given it is not a singular message?

File: src/kafka/kafka_hello_world.clj [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+                           "key.deserializer"   StringDeserializer
+                           "value.deserializer" StringDeserializer}}))
+
+
+
+(defn build-producer
+  []
+  (KafkaProducer. (get @properties :producer)))
+
+(defn build-consumer
```
Comment (lukaszhall) [inline src/kafka/kafka_hello_world.clj:30]: Why not idempotent?

File: src/kafka/kafka_hello_world.clj [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+                           "value.serializer"   StringSerializer}
+         :consumer-config {"client.id",         (.getHostName (InetAddress/getLocalHost))
+                           "group.id",          "group1"
+                           "bootstrap.servers", "localhost:9092"
+                           "key.deserializer"   StringDeserializer
+                           "value.deserializer" StringDeserializer}}))
+
+
+
+(defn build-producer
```
Comment (lukaszhall) [inline src/kafka/kafka_hello_world.clj:24]: Why not idempotent?

File: src/kafka/kafka_hello_world.clj [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+  []
+  (KafkaProducer. (get @properties :producer)))
+
+(defn build-consumer
+  [topic]
+  (let [consumer         (KafkaConsumer. (get @properties :consumer))
+        _subscribe-topic (.subscribe consumer [topic])]
```
Comment (lukaszhall) [inline src/kafka/kafka_hello_world.clj]: if this is `nil`, don't give it name

File: src/kafka/kafka_hello_world.clj [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+                           "group.id",          "group1"
+                           "bootstrap.servers", "localhost:9092"
+                           "key.serializer"     StringSerializer
+                           "value.serializer"   StringSerializer}
+         :consumer-config {"client.id",         (.getHostName (InetAddress/getLocalHost))
+                           "group.id",          "group1"
+                           "bootstrap.servers", "localhost:9092"
+                           "key.deserializer"   StringDeserializer
+                           "value.deserializer" StringDeserializer}}))
+
```
Comment (lukaszhall) [inline src/kafka/kafka_hello_world.clj:21]: Should (also) have an atom for kafka clients, as well as functions to that will populate the atoms with clients

File: src/kafka/kafka_hello_world.clj [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+  (deref producer-consumer)
+  #_=> #_[#object[org.apache.kafka.clients.producer.KafkaProducer
+                  0x301c93e1
+                  "org.apache.kafka.clients.producer.KafkaProducer@301c93e1"]
+          #object[org.apache.kafka.clients.consumer.KafkaConsumer
+                  0x4f2f21e6
+                  "org.apache.kafka.clients.consumer.KafkaConsumer@4f2f21e6"]]
+
+
+  (send-msg! (build-producer) "Message content")
```
Comment (lukaszhall) [inline src/kafka/kafka_hello_world.clj]: Building a producer every time you send a message? See above for saving state in atoms

File: src/kafka/kafka_hello_world.clj (hunk 9 above)
Comment (LucyxxNie) [reply src/kafka/kafka_hello_world.clj:21]: Should I atomize the my clients when I build them?

**Pull request merged**